### PR TITLE
kubespray: allow to manage ci labels with /label

### PR DESCRIPTION
**Pull request by VannTen**
Follow-up to #33289, to enable the ci-* labels to be managed by org-members with /label commands

I'm not sure this is the correct way to enable label management for org members(kubernetes-sigs), if not, I'll update the PR.

@BenTheElder following the thread on slack in #sig-testing

/cc @yankay @ant31 @tico88612

### CODE REVIEW

**Review by tico88612 — APPROVED** [review]
/lgtm
/assign @dims
Could you please take a look? Thanks!

**Comment by BenTheElder** [comment]
cc @kubernetes/sig-contributor-experience-leads

**Comment by Priyankasaggu11929** [comment]
Hello @VannTen, trying to understand – are you looking for just (i) the ability to add `ci-full/short/extended` labels using the label plugin (like with `/label ci-short`), OR do you also want (ii) to restrict this to Kubernetes-sigs org members?

If it's just (i) - PR https://github.com/kubernetes/test-infra/pull/21679 is an example on how to do it.

If both (i) + (ii), that will affect the entire `kubernetes-sigs` contributor workflow and require dedicated discussion within SIG ContribEx.

But IIUC, the `ci-full/short/extended` labels are only available in `kubespray` repository - so, adding these labels in [`additional_labels` section in plugins.yaml](https://github.com/kubernetes/test-infra/blob/ebbbbc8a2512d17f791736952a300a45a0e9a6a0/config/prow/plugins.yaml#L161-L172) here should help.

**Comment by VannTen** [comment]
My initial thinking was (i) + (ii), following the intent expressed in #33289 (to me "approvers or reviewers or members" reads as org members -> kubernetes-sigs)

However, I was not part of that first PR (holidays :) ) so maybe @tico88612 and @ant31 could chime in to express the original intent (aka, who should have the ability to request those labels exactly) ?

> If both (i) + (ii), that will affect the entire `kubernetes-sigs` contributor workflow and require dedicated discussion within SIG ContribEx.

Do you mean that this affect kubernetes-sigs contributor when/if they contribute to kubespray, or contributors on any repo under kubernetes-sigs/ ?
If it's the latter, does that mean that the labels restrictions are global and not possible per repository ?

In any case, I don't think the intent was to affect anyone outside kubespray, so we might settle on (i).
Just to be confirm, that mean than anyone (kubernetes-sigs member or not) could use /label to set those labels, correct ?

**Comment by Priyankasaggu11929** [comment]
> Do you mean that this affect kubernetes-sigs contributor when/if they contribute to kubespray, or contributors on any repo under kubernetes-sigs/ ?

I meant the latter – `contributors on any repo under kubernetes-sigs/`.
 
Because, [the current changes](https://github.com/kubernetes/test-infra/pull/33576/files#diff-f388c13accc6d3312ef9fb1970eedccddade2ce24b6647e9349a3ddc8dc37137R226-R235) in this PR won't work to restrict the application of `/label ci-short/full/extended` to all kubernetes-sigs org members.
With the current config changes, we can only restrict the label application to GH teams or GH users
(basically `allowed_teams` field needs valid GH teams OR valid GH usernames and we don't have a `kubernetes-sigs` GH team).

If you think, the `/label ci-short/full/extended` should be restricted to a small set of folks (like kubespray maintainers OR approvers OR reviewers, etc), then we can create GH teams for them and then add those teams to  `allowed_teams` field.

IMU, Global label restrictions (i.e. restriction of label plugin usage, to all org members) are handled within the labels plugin code itself atm (example - https://github.com/kubernetes/test-infra/pull/19272) and that's why it changes the entire label plugin workflow and we don't want to do that unless there's an absolute requirement, and after community consensus.

> In any case, I don't think the intent was to affect anyone outside kubespray, so we might settle on (i).
> Just to be confirm, that mean than anyone (kubernetes-sigs member or not) could use /label to set those labels, correct ?

correct.

**Comment by VannTen** [comment]
Ok, thanks for the explanation :+1:

In light of this IMO we have two choices (we're not going to rock the whole boat):

- A. Restrict the ci-* labels to kubespray teams (I see we currently have -maintainers and -admins)
- B. Allow anyone to apply the ci-* labels.

I would be inclined to go with B:
- the currently defined are not enough, IMO, so we would need to add more (-reviewers ?) if we were to go with A.
- there is already /ok-to-test to protect against excessive/malicious CI triggers (and for that matter, the motivated attacker could rewrite the jobs definition in the pull request anyway)

What do you think @ant31 @mzaian @yankay @tico88612 ?

**Comment by mzaian** [comment]
> Ok, thanks for the explanation 👍 In light of this IMO we have two choices (we're not going to rock the whole boat): - A. Restrict the ci-* labels to kubespray teams (I see we currently have -maintainers and -admins) - B. Allow anyone to apply the ci-* labels. I would be inclined to go with B: - the currently defined are not enough, IMO, so we would need to add more (-reviewers ?) if we were to go with A. - there is already /ok-to-test to protect against excessive/malicious CI triggers (and for that matter, the motivated attacker could rewrite the jobs definition in the pull request anyway) What do you think @ant31 @mzaian @yankay @tico88612 ?

I agree with A and not B.

**Comment by mzaian** [comment]
> I agree with A and not B.
> Any specific reasons ?
I'm actually in-favour for not allowing ok-to-test to org members. But this is not in our hands. If you think allowing the ci labels will have no impact then it's ok from my side.

**Comment by ant31** [comment]
I tend to prefer A in general, but I would go for B for now

I see two main scenarios:
1. Someone do ci-short  for a complex PR and we miss test cases 
  -> Main branch will fail, so we won't "miss" a bug. PR is reverted. 
2. Someone do ci-extended on many PRs to be annoying 
  -> The CI cluster is not elastic anymore, so it won't cost anything. Some manual work to clear the queue
  
 We can always limit to kubespray maintainer later if something happen.

**Comment by yankay** [comment]
I think Plan B is OK because it is more new developer friendly.

**Comment by VannTen** [comment]
Updated the PR to implement B.

/retitle kubespray: allow to manage ci labels with /label

**Comment by Priyankasaggu11929** [comment]
Tested label plugin is working as intended – https://github.com/kubernetes-sigs/kubespray/pull/11619#issuecomment-2419432481

**Comment by VannTen** [comment]
Thanks !

**Pull request merged**
